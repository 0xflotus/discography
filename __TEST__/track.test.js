import testSnapshot from './fixtures/testSnapshot';

test('Should implement tracks artist connection', async () => {
  const query = `
    query {
      lookup {
        release(id: 3) {
          tracks (first: 1) {
            edges {
              node {
                trackno
                title
                artists {
                  edges {
                    node {
                      id
                    }
                  }
                }
              }
            }
          }
        }
      }
    }
  `;

  await testSnapshot(query);
});

test('Should implement the track extra artist connection', async () => {
  const query = `
    query {
      lookup {
        release(id: 3) {
          tracks (first: 3) {
            edges {
              node {
                title
                trackno
                extraArtists {
                  edges {
                    node {
                      id
                    }
                  }
                }
              }
            }
          }
        }
      }
    }
  `;

  await testSnapshot(query);
});

test('Should paginate tracks by their track number', async () => {
  const query = `
    query {
      lookup {
        release(id: 1) {
          tracks (first:1, after:"MQ==") {
            pageInfo {
              hasNextPage
              endCursor
            }
            edges {
              node {
                title
                trackno
              }
              cursor
            }
          }
        }
      }
    }
  `;

  await testSnapshot(query);
});
