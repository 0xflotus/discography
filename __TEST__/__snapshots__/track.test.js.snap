// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should implement the track extra artist connection 1`] = `
Object {
  "data": Object {
    "lookup": Object {
      "release": Object {
        "tracks": Object {
          "edges": Array [
            Object {
              "node": Object {
                "extraArtists": Object {
                  "edges": Array [],
                },
                "title": "Untitled 8",
                "trackno": 1,
              },
            },
            Object {
              "node": Object {
                "extraArtists": Object {
                  "edges": Array [],
                },
                "title": "Anjua (Sneaky 3)",
                "trackno": 2,
              },
            },
            Object {
              "node": Object {
                "extraArtists": Object {
                  "edges": Array [
                    Object {
                      "node": Object {
                        "id": 8,
                      },
                    },
                  ],
                },
                "title": "When The Funk Hits The Fan (Mood II Swing When The Dub Hits The Fan)",
                "trackno": 3,
              },
            },
          ],
        },
      },
    },
  },
}
`;

exports[`Should implement tracks artist connection 1`] = `
Object {
  "data": Object {
    "lookup": Object {
      "release": Object {
        "tracks": Object {
          "edges": Array [
            Object {
              "node": Object {
                "artists": Object {
                  "edges": Array [
                    Object {
                      "node": Object {
                        "id": 4,
                      },
                    },
                    Object {
                      "node": Object {
                        "id": 5,
                      },
                    },
                  ],
                },
                "title": "Untitled 8",
                "trackno": 1,
              },
            },
          ],
        },
      },
    },
  },
}
`;

exports[`Should paginate tracks by their track number 1`] = `
Object {
  "data": Object {
    "lookup": Object {
      "release": Object {
        "tracks": Object {
          "edges": Array [
            Object {
              "cursor": "Mg==",
              "node": Object {
                "title": "Vasastaden",
                "trackno": 2,
              },
            },
          ],
          "pageInfo": Object {
            "endCursor": "Mg==",
            "hasNextPage": false,
          },
        },
      },
    },
  },
}
`;
